thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java

```diff
@@ -1,250 +1,445 @@
-            thirdEyeResponse, CURRENT_VALUE);
-      }
+
+  private static String TIME_DIMENSION_JOINER_ESCAPED = "\\|";
+  private static String TIME_DIMENSION_JOINER = "|";
+  private double metricThreshold = CollectionConfig.DEFAULT_THRESHOLD;
+  private static String OTHER = "OTHER";
+
+  public TimeOnTimeResponseParser(ThirdEyeResponse baselineResponse,
+      ThirdEyeResponse currentResponse, List<Range<DateTime>> baselineRanges,
+      List<Range<DateTime>> currentRanges, TimeGranularity timeGranularity,
+      List<String> groupByDimensions) {
+    this.baselineResponse = baselineResponse;
+    this.currentResponse = currentResponse;
+    this.baselineRanges = baselineRanges;
+    this.currentRanges = currentRanges;
+    this.aggTimeGranularity = timeGranularity;
+    this.groupByDimensions = groupByDimensions;
+
+    String collection = baselineResponse.getRequest().getCollection();
+    try {
+      collectionConfig = CACHE_REGISTRY_INSTANCE.getCollectionConfigCache().get(collection);
+    } catch (Exception e) {
+      LOGGER.debug("No collection configs for collection {}", collection);
     }
-    return metricGroupByDimensionData;
-  }
 
-  public static Map<MetricFunction, Map<String, Double>> processAggregateResponse(
-      Map<ThirdEyeRequest, ThirdEyeResponse> queryResultMap) {
-    // Map<metricFunction,Map<(baseline|current), Double>>>
-    Map<MetricFunction, Map<String, Double>> metricOnlyData = new HashMap<>();
-
-    for (Entry<ThirdEyeRequest, ThirdEyeResponse> entry : queryResultMap.entrySet()) {
-      ThirdEyeRequest thirdEyeRequest = entry.getKey();
-      ThirdEyeResponse thirdEyeResponse = entry.getValue();
-      String requestReference = thirdEyeRequest.getRequestReference();
-      if (ALL_BASELINE.equals(requestReference)) {
-        parseMetricOnlyResponse(metricOnlyData, thirdEyeResponse, BASELINE_VALUE);
-      }
-      if (ALL_CURRENT.equals(requestReference)) {
-        parseMetricOnlyResponse(metricOnlyData, thirdEyeResponse, CURRENT_VALUE);
-      }
+    if (collectionConfig != null) {
+      metricThreshold = collectionConfig.getMetricThreshold();
     }
-    return metricOnlyData;
   }
 
-  public static void parseMetricOnlyResponse(Map<MetricFunction, Map<String, Double>> aggregateData,
-      ThirdEyeResponse thirdEyeResponse, String baselineOrCurrent) {
-    List<MetricFunction> metricFunctions = thirdEyeResponse.getMetricFunctions();
-    for (MetricFunction metricFunction : metricFunctions) {
-      Map<String, String> row = thirdEyeResponse.getRow(metricFunction, 0);
-      // TODO THIS DEPENDS STRICTLY ON THE ORDER OF THE KEYS RETURNED?
-      String baselineValue = row.values().iterator().next();
-      if (!aggregateData.containsKey(metricFunction)) {
-        aggregateData.put(metricFunction, new HashMap<String, Double>());
-      }
-      aggregateData.get(metricFunction).put(baselineOrCurrent, Double.parseDouble(baselineValue));
+  List<Row> parseResponse() {
```

Comment:
extract each if/else block to a separate methods such as parseGroupByTimeResponse, parseGroupByTimeDimensionResponse parseGroupByDimensionResponse